replica/table.cc

```diff
@@ -2351,6 +2356,15 @@ table::table(schema_ptr schema, config config, lw_shared_ptr<const storage_optio
     set_metrics();
 }
 
+void table::on_flush_timer() {
+    tlogger.debug("on table {}.{} flush timer, period {}ms", _schema->ks_name(), _schema->cf_name(), _schema->memtable_flush_period());
+    (void)with_gate(_async_gate, [this] {
+        return flush().finally([this] {
+            _flush_timer.rearm(timer<>::clock::now() + std::chrono::milliseconds(_schema->memtable_flush_period()));
+        });
```

Comment:
we always can consider that synchronous code blocks seastar executes atomically?

Reply:
Speaking roughly every async call switches seastar thread execution context but sync call never does it?
So if we have a code block instead of `std::this_thread::sleep_for(1s)` in this example it would be executed atomically when other concurrent (async) operations will stay awaiting

Reply:
> Speaking roughly every async call switches seastar thread execution context but sync call never does it? So if we have a code block instead of `std::this_thread::sleep_for(1s)` in this example it would be executed atomically while other concurrent (async) operations will stay awaiting

Yes, the reason for that is single thread/process per core and cooperative multitasking. A sync call or whatever means reactor doesn't have a chance to run (so no further tasks can be scheduled). The running task must voluntarily yield the cpu. If a task does, `while(true) { volatile int x = 10 * 10; }`, no other task will ever have a chance to run. Seastar is responsible for scheduling its tasks, not Linux.